rules/manager.go

```diff
@@ -226,9 +229,9 @@ func (g *Group) run(ctx context.Context) {
 	iter := func() {
 		iterationsScheduled.Inc()
 
-		start := time.Now()
+		g.SetEvaluationTimestamp(time.Now())
 		g.Eval(ctx, evalTimestamp)
-		timeSinceStart := time.Since(start)
+		timeSinceStart := time.Since(g.evaluationTimestamp)
```

Comment:
While I don't see that a single group can be executed twice concurrently, you're using the mutex-protected function to set `g.evaluationTimestamp`, but then accessing it directly here without mutex to read it back.

IMO I would just keep the `start` variable as it is and additionally add the `g.SetEvaluationTimestamp(start)` call.

One thing to think about: do you really want to already set the evaluation timestamp *before* the rule evaluation happens or only after it completes? We only set the eval duration afterwards (naturally), and probably it'd be less confusing to also set the timestamp only afterwards?

Reply:
I'll add the start variable back. The idea with adding the timestamp prior to to completing the evaluation was to represent the last time that rules were attempted to be evaluated (at the start of evaluation). I think you're right that it would be more consistent to set the timestamp afterwards, though.

I think other monitoring tools (e.g. Nagios) also use the check (i.e. rule evaluation) completion time rather than start time to signify the last evaluation, so it'd be more consistent for people coming from other products, too.